<launch>

  <arg name="INPUT_PEOPLE_POSE" default="/edgetpu_human_pose_estimator/output/poses" />
  <arg name="INPUT_DEPTH_IMAGE" default="/head_camera/depth_registered/hw_registered/image_rect" />
  <arg name="INPUT_CAMERA_INFO" default="/head_camera/rgb/camera_info" />
  <arg name="depth_threshold" default="1.5"/>

  <!-- filter background as mask based on depth_threshold -->
  <group ns="welcome_to_jsk" >

    <node name="depth_image_filter"
          pkg="jsk_perception" type="depth_image_filter.py"
          output="screen" respawn="true">
      <remap from="~input" to="$(arg INPUT_DEPTH_IMAGE)" />
      <remap from="~output/mask" to="$(arg INPUT_DEPTH_IMAGE)/mask" />
      <rosparam subst_value="true" >
        threshold: $(arg depth_threshold)
      </rosparam>
    </node>

    <!-- use filtered mask to extract foreground rgb image -->
    <node name="apply_mask_image"
          pkg="nodelet" type="nodelet"
          args="standalone jsk_perception/ApplyMaskImage">
      <remap from="~input" to="$(arg INPUT_DEPTH_IMAGE)" />
      <remap from="~input/mask" to="$(arg INPUT_DEPTH_IMAGE)/mask" />
      <remap from="~output" to="$(arg INPUT_DEPTH_IMAGE)/depth_filtered" />
      <remap from="~output/mask" to="$(arg INPUT_DEPTH_IMAGE)/depth_filtered/mask" />
      <rosparam>
        approximate_sync: true
        clip: false
        cval: 0
      </rosparam>
    </node>

    <node name="people_pose_2d_to_3d"
          pkg="welcome_to_jsk_fetch" type="people_pose_2d_to_3d.py"
          output="screen" >
      <remap from="~input/pose" to="$(arg INPUT_PEOPLE_POSE)" />
      <remap from="~input/info" to="$(arg INPUT_CAMERA_INFO)" />
      <remap from="~input/depth" to="$(arg INPUT_DEPTH_IMAGE)/depth_filtered" />
      <rosparam subst_value="true">
        approximate_sync: true
        queue_size: 30
      </rosparam>
    </node>

    <node name="people_poses_to_poses"
          pkg="jsk_recognition_msgs" type="people_pose_array_to_pose_array.py">
      <remap from="~input" to="people_pose_2d_to_3d/output/pose" />
    </node>

    <node name="people_pose_aggregated_detector"
          pkg="welcome_to_jsk_fetch" type="people_pose_aggregated_detector.py">
      <remap from="~input" to="people_pose_2d_to_3d/output/pose" />
    </node>

  </group>

</launch>
